domain_config_ui/src/Config/ConfigFactory.php

```diff
@@ -0,0 +1,160 @@
+   *
+   * @var DomainConfigUIManager
+   */
+  protected $domainConfigUIManager;
+
+  /**
+   * Helper to check if config is allowed to be saved for domain.
+   *
+   * @param string $name
+   */
+  protected function isAllowedDomainConfig($name) {
+    // Get default allowed config and allow other modules to alter.
+    $allowed = $this->allowedDomainConfig;
+    \Drupal::moduleHandler()->alter('domain_config_allowed', $allowed);
+
+    // Return original name if reserved not allowed.
+    $is_allowed = FALSE;
+    foreach ($allowed as $config_name) {
+      // Convert config_name into into regex.
+      // Escapes regex syntax, but keeps * wildcards.
+      $pattern = '/^' . str_replace('\*', '.*', preg_quote($config_name, '/')) . '$/';
+      if (preg_match($pattern, $name)) {
+        $is_allowed = TRUE;
+      }
+    }
+
+    return $is_allowed;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\Core\Config\ConfigFactory::createConfigObject()
+   */
+  protected function createConfigObject($name, $immutable) {
+    if (!$immutable && $this->isAllowedDomainConfig($name)) {
+      $config = new Config($name, $this->storage, $this->eventDispatcher, $this->typedConfigManager);
+      // Pass the UI manager to the Config object.
+      $config->setDomainConfigUIManager($this->domainConfigUIManager);
+      return $config;
+    }
+    return parent::createConfigObject($name, $immutable);
+  }
+
+  /**
+   * Set the Domain config UI manager.
+   *
+   * @param DomainConfigUIManager $domain_config_ui_manager
+   */
+  public function setDomainConfigUIManager($domain_config_ui_manager) {
+    $this->domainConfigUIManager = $domain_config_ui_manager;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\Core\Config\ConfigFactory::doLoadMultiple()
+   */
+  protected function doLoadMultiple(array $names, $immutable = TRUE) {
+    // Let parent load multiple load as usual.
+    $list = parent::doLoadMultiple($names, $immutable);
+
+    // Do not apply overrides if configuring 'all' domains or config is immutable.
+    if (empty($this->domainConfigUIManager) || !$this->domainConfigUIManager->getSelectedDomainId() || !$this->isAllowedDomainConfig(current($names))) {
+      return $list;
+    }
+
+    // Pre-load remaining configuration files.
+    if (!empty($names)) {
+      // Initialise override information.
+      $module_overrides = [];
+      $storage_data = $this->storage->readMultiple($names);
+
+      // Load module overrides so that domain specific config is loaded in admin forms.
+      if (!empty($storage_data)) {
+        // Only get domain overrides if we have configuration to override.
+        $module_overrides = $this->loadDomainOverrides($names);
+      }
+
+      foreach ($storage_data as $name => $data) {
+        $cache_key = $this->getConfigCacheKey($name, $immutable);
+
+        if (isset($module_overrides[$name])) {
+          $this->cache[$cache_key]->setModuleOverride($module_overrides[$name]);
```

Comment:
I've applied the domain overrides on immutable config so that unconventional admin pages (like theme settings) show the domain domain specific config